### PR TITLE
[SPARK-22237] [CORE] Fix spark submit file download for standalone client mode

### PR DESCRIPTION
**Pull request by loneknightpy**
## What changes were proposed in this pull request?

This PR makes spark-submit script to use downloaded files in local/standalone client mode.

## How was this patch tested?

Unit tests and manual tests.

### CODE REVIEW

**Comment by gatorsmile** [comment]
cc @yhuai @jiangxb1987

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala [hunk 1]
```diff
@@ -366,6 +366,16 @@ object SparkSubmit extends CommandLineUtils with Logging {
       localPyFiles = Option(args.pyFiles).map {
         downloadFileList(_, targetDir, sparkConf, hadoopConf, secMgr)
       }.orNull
+
+      if (clusterManager == STANDALONE || clusterManager == LOCAL) {
+        // Use local files for standalone client mode.
+        args.primaryResource = localPrimaryResource
```
Comment (vanzin) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]: Isn't this handled by the code in L522 already?

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala [hunk 2]
```diff
@@ -366,6 +366,16 @@ object SparkSubmit extends CommandLineUtils with Logging {
       localPyFiles = Option(args.pyFiles).map {
         downloadFileList(_, targetDir, sparkConf, hadoopConf, secMgr)
       }.orNull
+
+      if (clusterManager == STANDALONE || clusterManager == LOCAL) {
```
Comment (vanzin) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]: Seems to me like what you want here is `clusterManager != YARN` instead.

**Review by vanzin — COMMENTED** [review]
Please remove the boilerplate code from the PR summary.

**Comment by vanzin** [comment]
@jerryshao

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala (hunk 1 above)
Comment (loneknightpy) [reply core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]: It doesn't seem so.

Comment (vanzin) [reply core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]: Can you explain what breaks without this line?

The code in L522 is adding the local copy of the primary resource to the user's classpath, which should be all that's needed, right?

**Comment by vanzin** [comment]
Also please add tests, otherwise this will be broken again.

**Comment by jerryshao** [comment]
Would you please show us an example of how it breaks? The codes here which assigning all resources to local ones might work, but it covers which line is really broken, can you please describe more? Thanks!

**Comment by vanzin** [comment]
> here with this change all the resources should be fetched from local driver

That's a good point. You should download resources just to add them to the driver's classpath, but executors can download them directly from the source.

As I requested, a test case would really help in understanding what is broken here.

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala (hunk 1 above)
Comment (loneknightpy) [reply core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]: Thanks for your input. It seems the issue is caused by a conflict with our internal patch. I will close this PR for now.

**Pull request closed without merging**